terraform/modules/providers/aws/virtual-machine-storage/storage.tf

```diff
@@ -0,0 +1,46 @@
+resource "aws_ebs_volume" "volume" {
+  count             = length(data.aws_instance.instances)
+  availability_zone = element(data.aws_instance.instances.*.availability_zone, count.index)
+  size              = var.vms_size
+  encrypted         = var.vms_encrypted
+  type              = var.vms_type
+
+  tags = {
+    Name            = "${var.vms_name}-${count.index}"
+    OwnerList       = var.vms_owner
+    ProjectList     = var.vms_project
+    EndDate         = var.vms_end_date
+    EnvironmentList = var.vms_env
+  }
+}
+
+resource "aws_volume_attachment" "volume_attachment" {
```

Comment:
Why Do we need an attached volume and an ebs volume on a host?
From Terraform docs

> NOTE on EBS block devices: If you use ebs_block_device on an aws_instance, Terraform will assume management over the full set of non-root EBS block devices for the instance, and treats additional block devices as drift. For this reason, ebs_block_device cannot be mixed with external aws_ebs_volume + aws_ebs_volume_attachment resources for a given instance.

Reply:
Due to how loops work on HCL, that's the only way I could find to allow the VM module spin up instances that don't need external volumes instances that need external storage.

@morrismukiri suggestions on doing it in a better way are welcome.

Reply:
FYI, I haven't used EBS block devices.

Reply:
Alright. I got you.